pdfminer/converter.py

```diff
@@ -723,3 +723,152 @@ def render(item: LTItem) -> None:
+        self.write("</head>\n")
+        self.write("<body>\n")
+
+    def write_footer(self):
+        self.write("<!-- comment in the following line to debug -->\n")
+        self.write("<!--script src='https://unpkg.com/hocrjs'>"
+                   "</script--></body>\n")
+
+    def write_text(self, text):
+        if self.stripcontrol:
+            text = self.CONTROL.sub("", text)
+        self.write(text)
+
+    def write_word(self):
+        if len(self.working_text) > 0:
+            bold_and_italic_styles = ''
+            if 'Italic' in self.working_font:
+                bold_and_italic_styles = 'font-style: italic; '
+            if 'Bold' in self.working_font:
+                bold_and_italic_styles += 'font-weight: bold; '
+            self.write("<span style='font:\"%s\"; font-size:%d; %s'"
+                       "class='ocrx_word' title='%s; x_font %s; "
+                       "x_fsize %d'>%s</span>" %
+                       ((self.working_font, self.working_size,
+                         bold_and_italic_styles,
+                         self.bbox_repr(self.working_bbox),
+                         self.working_font, self.working_size,
+                         self.working_text.strip())))
+        self.within_chars = False
+
+    def receive_layout(self, ltpage):
+
+        def render(item):
+            if self.within_chars and not isinstance(item, LTChar):
+                self.write_word()
+            if isinstance(item, LTPage):
+                self.page_bbox = item.bbox
+                self.write(
+                    "<div class='ocr_page' id='%s' title='%s'>\n" %
+                    (item.pageid, self.bbox_repr(item.bbox)))
+                for child in item:
+                    render(child)
+                self.write("</div>\n")
+            elif isinstance(item, LTTextLine):
+                self.write("<span class='ocr_line' title='%s'>" %
+                           ((self.bbox_repr(item.bbox))))
+                for child in item:
+                    render(child)
+                self.write("</span>\n")
+            elif isinstance(item, LTTextBox):
+                self.write(
+                    "<div class='ocr_block' id='%d' title='%s'>\n" %
+                    (item.index, self.bbox_repr(item.bbox)))
+                for child in item:
+                    render(child)
+                self.write("</div>\n")
+            elif isinstance(item, LTChar):
+                if not self.within_chars:
+                    self.within_chars = True
+                    self.working_text = item.get_text()
+                    self.working_bbox = item.bbox
+                    self.working_font = item.fontname
+                    self.working_size = item.size
+                else:
+                    if len(item.get_text().strip()) == 0:
+                        self.write_word()
+                        self.write(item.get_text())
```

Comment:
The layout algorithm inserts `LTAnno` every time it thinks a new word starts. The occurence of an `LTAnno` should be used to start a new word imo.